src/lib/symbol_overrides.cc

```diff
@@ -103,6 +113,16 @@ class DeallocTrackerHelper {
     }
   }
 
+  // disallow allocation during tracking
+  void track_no_alloc(void *ptr) {
+    if (_guard) {
+      tl_state()->allocation_allowed = false;
```

Comment:
is the reason we do this with a tl_state to help with testing ? 
Set the flag and check if any allocs are happening in the function ?

Reply:
which alternative had you in mind ?